src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -119,17 +140,25 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
                 long? resultId,
                 bool useVSCompletionItem,
                 CompletionTrigger completionTrigger,
-                Dictionary<ImmutableArray<CharacterSetModificationRule>, ImmutableArray<string>> commitCharacterRulesCache)
+                Dictionary<ImmutableArray<CharacterSetModificationRule>, ImmutableArray<string>> commitCharacterRulesCache,
+                bool returnTextEdits,
+                SourceText? documentText = null,
+                TextSpan? defaultSpan = null,
+                LSP.Range? defaultRange = null)
```

Comment:
i would prefer we *not* use optional arguments on internal apis.  it just means we can accidentally forget to pass this, or updat eall callers.  it would be better if these were non optional and then we were explicit at all callsites.